src/applications/vaos/new-appointment/newAppointmentFlow.js

```diff
@@ -61,17 +61,26 @@ function getFacilityPageKey(state) {
 }
 
 async function vaFacilityNext(state, dispatch) {
-  let eligibility;
+  let eligibility = getEligibilityStatus(state);
 
   if (selectUseFlatFacilityPage(state)) {
-    const facility = getChosenFacilityInfo(state);
-    const siteId = getSiteIdFromFakeFHIRId(facility.id);
-    eligibility = await dispatch(checkEligibility(facility, siteId));
+    // Fetch eligibility if we haven't already
+    if (eligibility.direct === null && eligibility.request === null) {
```

Comment:
Won't `eligibility` be null if we haven't fetched it yet?

Reply:
`getEligibilityStatus()` ends up returning the result of `isEligible()`, which always returns an object.  If no eligibility is found, it will return the snippet below:

```
  if (!eligibilityChecks) {
    return {
      direct: null,
      request: null,
    };
  }
```